src/main/java/jag/Ui.java

```diff
@@ -0,0 +1,288 @@
+        System.out.println(this.dashed);
+        System.out.println(response);
+        System.out.println(this.dashed);
+        command = scanner.nextLine();
+    }
+
+    /**
+     * Returns the description of the task from user input
+     *
+     * @param type Type represents T, D, E, F, which helps to indicate
+     *             the type of Task to be processed.
+     * @return description of the task from the user input on a case
+     *          by case basis, of T, D, E, F
+     */
+    public String getDescription(char type) {
+        String description = "";
+
+        if (type == 'T') {
+            String[] split = command.split("todo");
+            description = split[1].trim();
+        } else if (type == 'D') {
+            String[] split = command.split("/by");
+            description = split[0].replaceFirst("deadline", "").trim();
+        } else if (type == 'E') {
+            String[] split = command.split("/from | /to");
+            description = split[0].replaceFirst("event", "").trim();
+        } else {
+            String[] split = command.split("find");
+            description = split[1].trim();
+        }
+        return description;
+    }
+
+    /**
+     * Returns a String representation of the deadline from user input
+     * for a deadline task
+     *
+     * @return a String of the deadline set by the user for a deadline task
+     */
+    public String getBy() {
+        String[] split = command.split("/by");
+        return split[1].trim();
+    }
+
+    /**
+     * Returns a String representation of the starting date from user input
+     * for an event
+     *
+     * @return a String of the starting date set by the user for an event
+     */
+    public String getFrom() {
+        String[] split = command.split("/from | /to");
+        return split[1].trim();
+    }
+
+    /**
+     * Returns a String representation of the end date from user input
+     * for an event
+     *
+     * @return a String of the end date set by the user for an event
+     */
+    public String getTo() {
+        String[] split = command.split("/from | /to");
+        return split[2].trim();
+    }
+
+    /**
+     * Returns an Integer representing the index of the task to be deleted
+     * in the stored TaskList
+     *
+     * @return an Integer representing the index of the task to be deleted
+     *          in the stored TaskList
+     */
+    public int getDeleteIndex() {
+        char marker = command.charAt(command.length() - 1);
+        int index = 0;
+
+        if (Character.isDigit(marker)) {
+            index = Integer.parseInt(Character.toString(marker));
+        }
+        return index;
+    }
+
+    /**
+     * Returns an Integer representing the index of the task to be marked
+     *  or unmarked in the stored TaskList
+     *
+     * @return an Integer representing the index of the task to be marked
+     *          or unmarked in the stored TaskList
+     */
+    public int getMark() {
```

Comment:
Nit here, but could this be renamed to getMarkIndex so to more explain the named entity to the reader in more detail?